Implementing code coverage

suhaani-agarwal opened the pull request:
Implementing code coverage for animint2 using codecov. (using covr for R side and Chrome devtools coverage + v8-to-istanbul for JS side)

Closes #122

suhaani-agarwal commented:
code coverage is now being uploaded for both JS and R codes to codecov . see here : https://app.codecov.io/github/animint/animint2/commit/1747fde5868452bd9651a8637022bb3b75b84836/tree?dropdown=coverage

tdhock commented:
this is really excellent work, thanks!
I see files under vendor are used to compute code coverage for this report https://app.codecov.io/github/animint/animint2/commit/1747fde5868452bd9651a8637022bb3b75b84836/tree/inst/htmljs?dropdown=coverage

Can you please tell codecov to ignore the files under vendor/ ?

suhaani-agarwal commented on DESCRIPTION:
```diff
@@ -72,6 +72,7 @@ Depends:
     R (>= 3.4.0)
 Imports:
     servr,
+    covr,
```
okay, done!

suhaani-agarwal commented on .gitignore:
```diff
@@ -10,4 +10,5 @@
-.vscode/settings.json
\ No newline at end of file
+.vscode/settings.jsonnode_modules/
+node_modules/
```
Yes, I have now added node.js related files to .Rbuildignore

tdhock commented:
In CRAN test suite I still see
```
* checking top-level files ... NOTE
Non-standard file/directory found at top level:
  ‘v8-to-istanbul.js’
```
the fix is to add this file to Rbuildignore. But before doing that, can you please modify the CI so that the CRAN test suite will fail when there is a WARNING or NOTE?
so then we should get a red/failed build next.
and then you can push the fix after that?

suhaani-agarwal commented:
> In CRAN test suite I still see
> 
> ```
> * checking top-level files ... NOTE
> Non-standard file/directory found at top level:
>   ‘v8-to-istanbul.js’
> ```
> 
> the fix is to add this file to Rbuildignore. But before doing that, can you please modify the CI so that the CRAN test suite will fail when there is a WARNING or NOTE? so then we should get a red/failed build next. and then you can push the fix after that?

thanks for the clarification! I have now modified the CI so that the CRAN test suite will fail when there is a WARNING or NOTE: https://github.com/animint/animint2/actions/runs/16154589438/job/45594000311
and after that I fixed the error by adding the file v8-to-istanbul.js to Rbuildignore.

suhaani-agarwal commented:
The original [test-renderer2-PredictedPeaks](https://github.com/animint/animint2/blob/master/tests/testthat/test-renderer2-PredictedPeaks.R) was too heavy and was slowing down the browser, especially after we added JavaScript coverage collection through Chrome DevTools. I’ve now refactored the test to make it lighter by:
 - Subsetting the dataset to a minimal size,
 - Removing unnecessary visualization components not relevant to the test logic.

With this update, both JavaScript and R code coverage are now correctly collected and uploaded to Codecov:
https://app.codecov.io/github/animint/animint2/commit/9952830992b8e8f216647c4e174665234cd46bee/tree?dropdown=coverage

tdhock commented on tests/testthat/test-renderer2-PredictedPeaks.R:
```diff
@@ -8,17 +8,35 @@ writeBin(content(request), PredictedPeaks.RData)
+selected_chroms <- c("chr16", "chrM", "chrY")
+selected_types <- c("neutro", "mono")
+PredictedPeaks$chromCounts <- subset(PredictedPeaks$chromCounts, 
+                                   chrom %in% selected_chroms & 
+                                   type %in% selected_types & 
+                                   samples.up %in% c(1, 14, 38))
+PredictedPeaks$countsByChrom <- subset(PredictedPeaks$countsByChrom,
+                                     chrom %in% selected_chroms)
+PredictedPeaks$chrom.ranges <- subset(PredictedPeaks$chrom.ranges,
+                                    chrom %in% selected_chroms)
+PredictedPeaks$scatter.text <- subset(PredictedPeaks$scatter.text,
+                                    chrom %in% selected_chroms)
+PredictedPeaks$bg.rect <- subset(PredictedPeaks$bg.rect,
+                               chrom %in% selected_chroms & 
+                               nonInputType %in% selected_types &
+                               up %in% c(1, 14, 38))[1:50,]
```
can you please make a copy of the data set (instead of modifying the original), and change the indentation to below?
```r
SomePeaks <- PredictedPeaks
SomePeaks$bg.rect <- subset(
  SomePeaks$bg.rect,
  chrom %in% selected_chroms
)[1:50,]
```
etc?

tdhock commented on tests/testthat/test-renderer2-PredictedPeaks.R:
```diff
@@ -27,17 +45,17 @@ viz <- list(
+    geom_text(aes(relative.middle, type, label=samples.up,
+                href=paste0(
+                  "http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&position=",
+                  chrom, ":", zoomStart, "-", zoomEnd)),
```
if you want to reformat, please use indentation like this:
```r
    geom_text(aes(
      relative.middle, href=paste0(
        "http:")),
      data=SomePeaks$chromCounts)+
```
etc

suhaani-agarwal commented:
Apologies for any confusion caused. While updating NEWS.md, I accidentally ran git rebase instead of merging master into coverage, which resulted in some duplicate commits and a rewritten history due to a force push.

I've verified that all intended changes ,including test improvements and coverage additions are intact. The branch is now clean, up-to-date with master, and ready for review.
Please let me know if any further changes are needed

tdhock commented:
Can you please provide a link to a more recent coverage report which shows R/JS code coverage? (but not vendor files)

Like this: https://app.codecov.io/github/animint/animint2/commit/1747fde5868452bd9651a8637022bb3b75b84836/tree/inst/htmljs?dropdown=coverage

For some reason I can't find a newer report on codecov. What do I need to click on?

suhaani-agarwal commented:
> Can you please provide a link to a more recent coverage report which shows R/JS code coverage? (but not vendor files)
> 
> Like this: https://app.codecov.io/github/animint/animint2/commit/1747fde5868452bd9651a8637022bb3b75b84836/tree/inst/htmljs?dropdown=coverage

coverage report for the most recent commit : https://app.codecov.io/github/animint/animint2/commit/1065d7f9c082536e4db9e0e90dc7abd0717883ba/tree
Please click on "inst" -> "htmljs" -> "animint.js" to view the JS code coverage : https://app.codecov.io/github/animint/animint2/commit/1065d7f9c082536e4db9e0e90dc7abd0717883ba/blob/inst/htmljs/animint.js?dropdown=coverage

and on "R" -> "any file" to view the R code coverage for that file

> For some reason I can't find a newer report on codecov. What do I need to click on?

So the link to the homepage for the coverage reports for animint2 is : https://app.codecov.io/github/animint/animint2
here, after selecting the branch "coverage" , you will be able to access the most recent coverage report for different files (after scrolling down) : https://app.codecov.io/github/animint/animint2/tree/coverage

In order to view the coverage reports for different commits, you can click on "commits" in the navbar and then click on the commit you want to view the coverage report for. 

Shall I make a video demonstrating the usage of this dashboard?

tdhock commented:
ok, I had to click File Explorer.
Thanks!
Video would be nice for your blog.
Great work!!
